engine/evarutil.ml

```diff
@@ -100,6 +100,14 @@ let has_undefined_evars evd t =
   try let _ = f h t in false
   with (Not_found | NotInstantiatedEvar) -> true
 
+let has_undefined_evars_or_metas evd t =
+  let rec has_ev t =
+    match EConstr.kind evd t with
+    | Evar _ | Meta _ -> raise NotInstantiatedEvar
+    | _ -> EConstr.iter evd has_ev t in
+  try let _ = has_ev t in false
+  with (Not_found | NotInstantiatedEvar) -> true
```

Comment:
The Not_found is fishy, it would mean the term has an unknown evar.

Reply:
I think I copy-pasted the code from `has_undefined_evars`, which changed in the meantime, and it is out of the scope of this PR to fix this. Also, catching the unknown evars here is not a trivial design choice (although I agree that if we want the invariant that no evar may ever not be declared in the `evar_map` then we should probably explode as soon as possible). Should I mirror the code of `has_undefined_evars` again and open another PR to propose the change?